lib/node/asar-fs-wrapper.ts

```diff
@@ -857,13 +857,13 @@ export const wrapFsWithAsar = (fs: Record<string, any>) => {
   const { readdir } = fs;
   fs.readdir = function (pathArgument: string, options: ReaddirOptions, callback: ReaddirCallback) {
     callback = typeof options === 'function' ? options : callback;
-    validateFunction(callback, 'callback');
+    validateFunction(callback, 'callback')!;
```

Comment:
These are necessary because TS doesn't like the typings for these `validate*` functions from Node because you can't use `asserts x is type` syntax with arrow functions, see microsoft/TypeScript#34523. There's workarounds but the typings in Node are being done via JSDoc and not sure a workaround can be done in that particular case.